main.go

```diff
@@ -131,6 +131,15 @@ func cmd() *cobra.Command {
 					set(&context.UpgradeCodeMigration)
 				case "pf":
 					set(&context.UpgradePfVersion)
+				case "check-upstream-version":
+					if targetVersion != "" {
+						return fmt.Errorf(
+							"--kind check-upstream-version is incompatible with --target-version, cannot set both",
+						)
+					}
+					set(&context.UpgradeProviderVersion)
+					set(&context.OnlyCheckUpstream)
```

Comment:
it feels like this actually wants to be two (or even three) context flags: 
- `CheckUpstream` which triggers the logic to look at the upstream for versions
- `CreateIssue` which tries to create the issue if missing (maybe could be combined with `OnlyCheckUpstream` if we'd skip creating the issue when upgrading directly) 
- `OnlyCheckUpstream` which causes the command to return early after issue creation

Reply:
I hear you on this. My personal end goal is to remove issue creation altogether but if that turns out to be too ambitious, we can add extra flags.